Xamarin.Forms.Core/Binding.cs

```diff
@@ -113,9 +116,12 @@ internal override void Apply(bool fromTarget)
 		internal override void Apply(object newContext, BindableObject bindObj, BindableProperty targetProperty)
 		{
 			object src = _source;
-			base.Apply(src ?? newContext, bindObj, targetProperty);
+			if (newContext == null && src == null)
+				base.Apply(NullValue, bindObj, targetProperty);
```

Comment:
if `NulValue` belongs to `BindingBase`, you shouldn't have to handle it in `Binding` at all